.github/workflows/default.yml

```diff
@@ -77,26 +101,120 @@ jobs:
 
   typecheck-mypy:
+    name: Check typing and annotations
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
+    - name: Checkout the source code
+      uses: actions/checkout@v3
       with:
-        fetch-depth: 2
+        fetch-depth: 0
     - name: Set up Python
       uses: actions/setup-python@v4
       with:
-        python-version: "3.10"
+        python-version: ${{ env.PYTHON_LATEST }}
         check-latest: true
         cache: pip
         cache-dependency-path: |
           setup.py
           requirements-dev.txt
           requirements-doc.txt
-    - name: Install
+    - name: Install dependencies
       run: |
         pip install -U pip setuptools wheel
         pip install -U -r requirements-dev.txt
     - name: Typecheck with mypy
       run: |
         echo "::add-matcher::.github/workflows/mypy-matcher.json"
         python -m mypy aiomonitor/ examples/ tests/ setup.py
+
+  test-pytest:
+    name: Tests on ${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+    continue-on-error: ${{ matrix.experimental }}
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version:
+          - "3.8"
+          - "3.9"
+          - "3.10"
+          - "3.11"
+        experimental: [false]
+        include:
+          - python-version: "~3.12.0-0"
+            experimental: true
+    steps:
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Checkout the source code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Install dependencies
+        run: |
+          pip install --upgrade setuptools
+          pip install -r requirements-dev.txt
+      - name: Run tests
+        run: make test
+      - name: Upload coverage data
+        uses: codecov/codecov-action@v3
+        with:
+            token: ${{ secrets.CODECOV_TOKEN }}
+
+  check:  # This job does nothing and is only used for the branch protection
+    name: ✅ Ensure the required checks passing
+    if: always()
+    needs: [lint-isort, lint-flake8, lint-black, typecheck-mypy, test-pytest]
+    runs-on: ubuntu-latest
+    steps:
+    - name: Decide whether the needed jobs succeeded or failed
+      uses: re-actors/alls-green@release/v1
+      with:
+        jobs: ${{ toJSON(needs) }}
+
+  build-sdist:
+    name: 📦 Build the source distribution
+    runs-on: ubuntu-latest
+    if: github.event_name == 'release' && github.event.action == 'created'
+    steps:
+      - name: Checkout project
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.PYTHON_LATEST }}
+      - run: python -Im pip install build
+        name: Install core libraries for build and install
+      - name: Build artifacts
+        run: python -Im build
```

Comment:
Since the linting jobs currently don't depend on this one, maybe it's a good idea to run `twine check --strict` after this step. Alternatively, I'd put linting in a dedicated job.